spec/Section 2 -- Language.md

```diff
@@ -13,28 +13,44 @@ double-colon `::`).
 
 ## Source Text
 
-SourceCharacter :: /[\u0009\u000A\u000D\u0020-\uFFFF]/
+SourceCharacter :: "Any Unicode code point"
 
 GraphQL documents are expressed as a sequence of
-[Unicode](http://unicode.org/standard/standard.html) characters. However, with
-few exceptions, most of GraphQL is expressed only in the original non-control
-ASCII range so as to be as widely compatible with as many existing tools,
-languages, and serialization formats as possible and avoid display issues in
-text editors and source control.
+[Unicode](http://unicode.org/standard/standard.html) code points (referred to in
+this specification as characters). All Unicode code point values from U+0000 to
+U+10FFFF, including surrogate code points, may appear in this sequence where
+allowed by the grammatical rules below.
 
+A [combining character sequence](http://unicode.org/faq/char_combmark.html) is
+treated as a sequence of individual Unicode code points and a sequence of
+individual {SourceCharacter}, even though they may appear to a user as a
+single character.
 
-### Unicode
+However, with the exceptions of {StringValue} and {Comment}, most of GraphQL is
+expressed only in the original non-control ASCII range so as to be as widely
+compatible with as many existing tools, languages, and serialization formats as
+possible and avoid display issues in text editors and source control.
+
+Note: The encoding used to represent a GraphQL document source is irrelevant to
+this specification. A document is not required to be stored or transmitted in an
+encoding which can represent every Unicode code point. Instead, given any
+encoding format, and the range of code points which it can encode, GraphQL
+documents may consist of any of those code points.
 
-UnicodeBOM :: "Byte Order Mark (U+FEFF)"
 
-Non-ASCII Unicode characters may freely appear within {StringValue} and
-{Comment} portions of GraphQL.
+### Byte Order Mark
+
+UnicodeBOM :: "Byte Order Mark (U+FEFF)"
```

Comment:
Like surrogates, this code point is irrelevant after decoding the document's encoding to Unicode. It won't appear.